pkg/odo/cli/init/init.go

```diff
@@ -74,12 +67,15 @@ type InitOptions struct {
 }
 
 // NewInitOptions creates a new InitOptions instance
-func NewInitOptions(backends []params.ParamsBuilder, fsys filesystem.Filesystem, initClient _init.Client, prefClient preference.Client) *InitOptions {
-	return &InitOptions{
-		backends:         backends,
-		fsys:             fsys,
-		initClient:       initClient,
-		preferenceClient: prefClient,
+func NewInitOptions() *InitOptions {
+	return &InitOptions{}
+}
+
+func (o *InitOptions) SetDependencies(dependencies *dependencies.Dependency) {
+	o.dependencies = dependencies
+	o.backends = []params.ParamsBuilder{
+		params.NewFlagsBuilder(),
+		params.NewInteractiveBuilder(asker.NewSurveyAsker(), catalog.NewCatalogClient(dependencies.FS, dependencies.PreferenceClient)),
 	}
 }
```

Comment:
If `SetDependencies` does everything, then do we need `New<CMD>Options` method?

Reply:
I think `SetClientset`/`SetDependencies` is only for initializing the clients needed by a command. `New<CMD>Options` do much more than that. There could/would be command specific fields in `<CMD>Options` struct which can't be initialized with `SetClientset`. It's out of its scope.

@feloy correct me if I am wrong.

Reply:
Yes, right, we are removing the initialization of the clients here, but there could be other initializations (not depending on clients).

But I also agree that in most cases, this initialization will be empty.

For consistency, let's use the New<Cmd>Options() form, event it is doing nothing more than creating an empty struct.

Reply:
> For consistency, let's use the NewOptions() form, event it is doing nothing more than creating an empty struct.

+1. Similarly, we are going to have empty `SetClientset` for many commands as well.